R/select.R

```diff
@@ -125,12 +125,27 @@ select.data.frame <- function(.data, ...) {
   loc <- tidyselect::eval_select(expr(c(...)), .data)
   loc <- ensure_group_vars(loc, .data, notify = TRUE)
 
-  set_names(.data[loc], names(loc))
+  dplyr_col_select(.data, loc, names(loc))
 }
 
 
 # Helpers -----------------------------------------------------------------
 
+dplyr_col_select <- function(.data, loc, names = NULL) {
+  out <- .data[loc]
+  if (!inherits(out, "data.frame")) {
+    abort(c(glue("could not reconstruct data frame."),
```

Comment:
Should be capitalised and use "can't"

```suggestion
    abort(c(glue("Can't reconstruct data frame."),
```